### PR TITLE
CI with execution table

### PR DESCRIPTION
**Pull request by stuartmcalpine**
Basic CI now adds execution and dataset_alias entries.

Added a test for execution table.

Currently the dataset_alias table is not set up for joins, or the ability to query, do we want to add this functionality ?

### CODE REVIEW

**Comment by JoanneBogart** [comment]
Yes, we do need to add ability to return query information for dataset_alias and similarly for the dependency table.

**Comment by stuartmcalpine** [comment]
I've updated the query function to work for an arbitrary number of tables (as long as they can all join to the dataset table).

I haven't tested all cases but the bulk is implemented.

**Comment by stuartmcalpine** [comment]
@JoanneBogart this should be ready to review now. 

I think the querying should now be generic. Maybe we can come up with a few more test queries for the CI.

Still need to add dependencies to the CI tests but there is not a registrar for the dependencies table  currently

**Comment by JoanneBogart** [comment]
@stuartmcalpine Thanks.  I'll take a look Monday at the latest.   The code in the new PR #18 will write to the dependency table.

File: src/dataregistry/query.py
Comment (JoanneBogart) [inline src/dataregistry/query.py:275]: This might not work properly for the dependency table;  an onclause argument might be required. The dependency table now (in my `dependency` branch) links input datasets to an execution.   If someone asks for dependencies the SQL will have to look like

`select from dataset join dependency on execution_id  `

Maybe that's what sqlalchemy will do by default but it needs to be tried.

**Review by JoanneBogart — CHANGES_REQUESTED** [review]
The only changes requested currently have to do with the use of "assert", however I suspect that the modified definition of the dependency table in my dependency branch (connects input datasets to the execution in which they were used) might cause problems for your generic approach.

File: src/dataregistry/query.py
Comment (stuartmcalpine) [reply src/dataregistry/query.py]: Ah, thank you for pointing this out.

I've changed the asserts here to raise Exceptions, as they all relate to error checking in the user inputs

File: src/dataregistry/query.py
Comment (stuartmcalpine) [reply src/dataregistry/query.py:276]: ok, lets wait till your branch is merged into main and then we can test this branch further after those changes are applied.

**Comment by stuartmcalpine** [comment]
@JoanneBogart I think this is now compatible with the main branch and can be reviewed.

I'm not sure I have enough tests around the dependencies, maybe we can come up with some more?

File: tests/ci/create_test_entries.py
Comment (JoanneBogart) [inline tests/ci/create_test_entries.py:189]: A very minor point but - could we change the names to `bumped_dataset`, `bumped_dataset_2`, etc.?

**Review by JoanneBogart — APPROVED** [review]
There are some minor suggestions that have to do with naming, documentation and, in one case, preferred way to write a line of code. I'm marking this "approved", assuming you'll more or less implement the suggestions before merging. If there are any you object to we should discuss them.

**Comment by JoanneBogart** [comment]
Although I've approved the PR there are some minor suggestions I'd like you to look at.

The tests are ok as far as they go.  There are additional cases that need testing but that can be in a future PR.   These tests should include
- Including a configuration file for one of the execution entries. I have one I used for my testing which we could add to the repo
- Including one or more dataset entries with a `version_suffix` and then bumping the version of such a dataset.
- registering one or more datasets where `is_dummy` is False
Another one which may require extra work in query.py is to ask for dependencies of a dataset. I don't think your generic join mechanism will work in that case but it is something people will want to do.

File: src/dataregistry/registrar.py
Comment (stuartmcalpine) [reply src/dataregistry/registrar.py:235]: `v_fields = {"major": 1, "minor": 0, "patch": 0}
".".join(str(v_fields.values()))`

gives

`d.i.c.t._.v.a.l.u.e.s.(.[.1.,. .0.,. .0.].)`

**Pull request merged**
